src/telemetry/mod.rs

```diff
@@ -18,7 +18,413 @@
+    ) {
+        tokio::select! {
+            _output = Telemetry::data_send_loop(interface_name, period, communication_channel) => {debug!("data_send_loop ended")},
+            _ = kill_switch.recv() => {debug!("Kill switch triggered")},
+        }
+    }
+
+    async fn data_send_loop(
+        interface_name: String,
+        period: u64,
+        communication_channel: MpscSender<TelemetryPayload>,
+    ) {
+        let mut interval = interval(Duration::from_secs(period));
+        loop {
+            interval.tick().await;
+            send_data(&communication_channel, interface_name.clone()).await;
+        }
+    }
+
+    async fn set_enabled(&self, interface_name: &str, enabled: bool) {
+        debug!("set {interface_name} to enabled {enabled}");
+
+        self.telemetry_task_configs
+            .clone()
+            .write()
+            .await
+            .entry(interface_name.to_string())
+            .or_insert(Default::default())
+            .override_enabled = Some(enabled);
+    }
+
+    async fn unset_enabled(&self, interface_name: &str) {
+        debug!("unset {interface_name} enabled");
+
+        if let Some(telemetry_task_config) = self
+            .telemetry_task_configs
+            .clone()
+            .write()
+            .await
+            .get_mut(interface_name)
+        {
+            telemetry_task_config.override_enabled = None;
+        }
+    }
+
+    async fn set_period(&self, interface_name: &str, period: u64) {
+        debug!("set {interface_name} to period {period}");
+        self.telemetry_task_configs
+            .clone()
+            .write()
+            .await
+            .entry(interface_name.to_string())
+            .or_insert(Default::default())
+            .override_period = Some(period);
+    }
+
+    async fn unset_period(&self, interface_name: &str) {
+        debug!("unset {interface_name} period");
+
+        if let Some(telemetry_task_config) = self
+            .telemetry_task_configs
+            .clone()
+            .write()
+            .await
+            .get_mut(interface_name)
+        {
+            telemetry_task_config.override_period = None;
+        }
+    }
+
+    pub async fn telemetry_config_event(
+        &mut self,
+        interface_name: &str,
+        endpoint: &str,
+        data: &AstarteType,
+    ) {
+        match (endpoint, data) {
+            ("enable", AstarteType::Boolean(enabled)) => {
+                self.set_enabled(interface_name, *enabled).await;
+            }
+
+            ("enable", AstarteType::Unset) => {
+                self.unset_enabled(interface_name).await;
+            }
+
+            ("periodSeconds", AstarteType::LongInteger(period)) => {
+                self.set_period(interface_name, *period as u64).await;
+            }
+
+            ("periodSeconds", AstarteType::Integer(period)) => {
+                self.set_period(interface_name, *period as u64).await;
+            }
```

Comment:
Why are you receiving both integer and longinteger for the same endpoint? Shouldn't it be either one of them?